### DISCUSSION TITLE
`eval` not allowed in Middleware pages/_middleware. I don't understand this error.

### DISCUSSION THREAD

**Discussion by denik1981** [vercel/next.js#31070] [Help]
I'm getting an error at runtime labeled as `eval not allowed in Middleware pages/_middleware` but I have looked into the code and deep into node_modules to find that eval is not being used. So the eval next js is complaing here is the one webpack is generated at run-time compilation for client requests. How can I solve this? Is that we re not allowed to import other modules when using middleware?

**Comment by IRediTOTO** [comment]
me too :|

**Comment by ncphillips** [comment]
I've ran into this as well. It's a dealbreaker for middleware at the moment unfortunately. It's preventing us from using Apollo in our middleware.

https://github.com/vercel/next.js/issues/31694

**Reply by ncphillips** [reply]
This is no longer a dealbreaker. The apollo team has worked around the issue on their end.

https://github.com/apollographql/apollo-client/issues/9128

**Comment by j-mendez** [comment]
Try using the direct import file instead of a general index ex: `@lib/configs/somefile`
